src/main/java/org/example/HomeworkDao.java

```diff
@@ -5,7 +5,7 @@
 import java.sql.SQLException;
 
 public class HomeworkDao {
-    public void addHomework(Homework homework) {
+    public void addHomework(Homework homework) throws HomeworkException {
```

Comment:
remove `throws HomeworkException` as you already handle this exception in try-catch